app_dart/lib/src/service/build_status_provider.dart

```diff
@@ -122,6 +124,31 @@ class BuildStatusService {
     }
   }
 
+  /// This function gets called with the last 40 builds fo a given builder ordered
+  /// by creation time starting with the last one first. [tasks] has a list of tasks
+  /// retrieved from BuildBucket using RPCs and [log] is the AppEngine logger to be
+  /// able to logging messages.
+  Future<String> latestLUCIStatus(List<LuciTask> tasks, Logging log) async {
+    for (LuciTask task in tasks) {
```

Comment:
This looks at all LUCI builds in the number of commits * number of builders. We shouldn't be checking builds if they have a green build above them. Should we refactor calculateCumulativeStatus so it injects `tasks` and we can reuse that algorithm?

Reply:
That will add back the latency to ingest the data. I would prefer to optimize this later with its own data model.